docs/architecture/adr-071-crypto-v2-multi-curve.md

```diff
@@ -0,0 +1,597 @@
+# ADR 071: Cryptography v2- Multi-curve support
+
+## Change log
+
+* May 7th 2024: Initial Draft (Zondax AG: @raynaudoe @juliantoledano @jleni @educlerici-zondax @lucaslopezf)
+
+## Status
+
+DRAFT
+
+## Abstract
+
+This ADR proposes the refactoring of the existing Keyring module to support multiple cryptographic curves for signing and verification processes. With this update, we aim to facilitate the integration of new cryptographic curves through clean and simple interfaces. Additionally, support for Hardware Security Modules (HSM) is introduced as a complementary enhancement in this redesign.
+
+This ADR also introduces the capability to support remote signers. This feature will enable the nodes to interact with cryptographic signers that are not locally present on the system where the main app is running. This is particularly useful for scenarios where keys are managed in secure, remote environments or when leveraging cloud-based cryptographic services.
+
+
+## Introduction
+
+The introduction of multi-curve support in the cosmos-sdk cryptographic module offers significant advantages. By not being restricted to a single cryptographic curve, developers can choose the most appropriate curve based on security, performance, and compatibility requirements. This flexibility enhances the application's ability to adapt to evolving security standards and optimizes performance for specific use cases, helping to future-proofing the sdk's cryptographic capabilities.
+
+
+Special focus has been placed on the following key aspects:
+
+* modularity
+* extensibility
+* security
+* maintainability
+* developer experience
+
+The enhancements in this proposal not only render the ["Keyring ADR"](https://github.com/cosmos/cosmos-sdk/issues/14940) obsolete, but also encompass its key aspects, replacing it with a more flexible and comprehensive approach. Furthermore, the gRPC service proposed in the Keyring ADR can be easily implemented as a specialized HSM.
+
+
+### Glossary
+
+1. **Interface**: In the context of this document, "interface" refers to Go's interface.
+
+2. **Module**: In this document, "module" refers to a Go module.
+
+3. **Package**: In the context of Go, a "package" refers to a unit of code organization.
+
+## Context
+
+In order to fully understand the need for changes and the proposed improvements, it's crucial to consider the current state of affairs:
+
+* The Cosmos SDK currently lacks a comprehensive ADR for the cryptographic package.
+
+* If a blockchain project requires a cryptographic curve that is not supported by the current SDK, they are obligated to fork the SDK repository and make modifications. These modifications could potentially make the fork incompatible with future updates from the upstream SDK, complicating maintenance and integration.
```

Comment:
I don't believe this is right? People have added curves without forking.

Reply:
Ok, maybe 'obligated' is not the best word because you could manage to do it without a fork. Nevertheless, having lets say 'pluggable cryptography' is aligned with the modular philosophy. 

I'm sure that cases like that can exists because of software nature, but from our point of view there's no clear framework that allows and encourage devs to do so.